src/Container.php

```diff
@@ -67,12 +68,21 @@ public function __construct(
 	 * @template ID
 	 * @param class-string<ID> $id The ID of the class to retrieve.
 	 * @return ID
+	 * @throws ContainerException In case the ID could not be resolved or instantiated.
 	 *
 	 * Psalm expects $id to be a string, based on ContainerInterface.
 	 * @psalm-suppress MoreSpecificImplementedParamType
+	 *
+	 * PSR-11 containers declares to throw an un-throwable interface
+	 * (it does not extend Throwable), and Psalm does not accept it.
+	 * @psalm-suppress MissingThrowsDocblock
 	 */
 	public function get( $id ) {
-		return $this->container->get( $id );
+		try {
+			return $this->container->get( $id );
+		} catch ( \Throwable $e ) {
+			throw new ContainerException( $e->getMessage() );
```

Comment:
> I guess I'm still not sure when and why we're using PSR rather than League 🤔

League uses PSR, and their code works with both `psr/container` 1.* and 2.*. However, one of our libraries is stuck with 1.1 (I think it's Psalm), and we're using 1.1, which is the one that doesn't implement `Throwable`. [2.0 does](https://github.com/php-fig/container/blob/master/src/ContainerExceptionInterface.php).

> 1. The PSR-11 class doesn't throw objects implementing Throwable ... But then catching a Throwable here won't help with scenario 1. It will still be a problem.

It's the PSR-11 interface that doesn't implement `Throwable`. League's exceptions, as you noticed, are based on `RuntimeException`, which implements `Throwable`.

> So is this just to keep Psalm happy?

Yes.